src/rngs/mod.rs

```diff
@@ -47,34 +54,40 @@
 //!   be replaced in the next major release.
 //! - [`StdRng`] is a CSPRNG chosen for good performance and trust of security
 //!   (based on reviews, maturity and usage). The current algorithm is HC-128,
-//!   which is recommended by ECRYPT's eSTREAM project. Amortised performance
-//!   is (on modern x86_64 CPUs) roughly similar to [`SmallRng`], but memory
-//!   usage is much higher.
+//!   which is one of the recommendations by ECRYPT's eSTREAM project.
 //!
 //! The above PRNGs do not cover all use-cases; more algorithms can be found in
 //! the [`prng` module], as well as in several other crates. For example, you
 //! may wish a CSPRNG with significantly lower memory usage than [`StdRng`]
 //! while being less concerned about performance, in which case [`ChaChaRng`]
 //! is a good choice.
 //!
-//! As mentioned above, PRNGs require random seed data in order to produce
-//! random output. This is especially important for CSPRNGs, which are still
+//! One complexity is that the internal state of a PRNG must change with every
+//! generated number. For APIs this generally means a mutable reference to the
+//! state of the PRNG has to be passed around.
+//!
+//! A solution is [`ThreadRng`]. This is a thread-local implementation of
```

Comment:
You mean a solution to having to pass around mutable state is just to keep a non-local instance? Genius. (irony)

`ThreadRng` is more than that. Oh well, I suppose the introduction will do.

What I don't like is the swapping of the following sentences: it's raison d'être is to be fast and secure; the auto-seeding is just an implementation detail (as is the fact that it uses `StdRng`; I've even been wondering if the two need to use the same algorithm).

Reply:
> `ThreadRng` is more than that.

Yes, genius :smile:. What would you say is the big thing to choose `ThreadRng`?

I see it mostly as "I need random numbers, but don't want to know how it works, and don't want to see it back in my API".

Can you open an issue someday to explore your ideas for the future of `ThreadRng`. You mentioned it in several issues.

Reply:
Yes, I do have one idea regarding it. But maybe it should just be called `NonLocalRng`; that should help with all the confusion about what this has to do with threads :laughing: 

Jokes aside, I'm wondering whether we should consider memory usage too, and whether we should separate crypto- and non-crypo-rngs here (to further save memory in threads not requiring the former). I'll make a PR at some point.